R/bibfix.R

```diff
@@ -0,0 +1,56 @@
+#' @rdname bibstats
+#' @importFrom stringr str_replace_all str_replace fixed
+#' @importFrom bib2df bib2df df2bib
+#' @importFrom purrr map
+#' @export
+fix_bib <- function(bib_file) {
+  if (missing(bib_file)) bib_file <- find_bib()
+  bib <- suppressMessages(bib2df::bib2df(bib_file))
+  bib$PAGES <- stringr::str_replace_all(bib$PAGES,
+                                        stringr::fixed("{\\textendash}"), "--")
+  bib$AUTHOR <- purrr::map(bib$AUTHOR, function(x) {
+    x <- stringr::str_replace(x, "^", "{")
+    x <- stringr::str_replace(x, "\\{\\{", "{")
+    x <- stringr::str_replace(x, "$", "}")
+    x <- stringr::str_replace(x, "\\}\\}", "}")
+  })
+  bib$EDITOR <- purrr::map(bib$EDITOR, function(x) {
+    x <- stringr::str_replace(x, "^", "{")
+    x <- stringr::str_replace(x, "\\{\\{", "{")
+    x <- stringr::str_replace(x, "$", "}")
+    x <- stringr::str_replace(x, "\\}\\}", "}")
+  })
+  bib2df::df2bib(bib, bib_file)
+}
```

Comment:
This function can be extended. It's mainly for how Zotero-RStudio-Bibtex converts everything. Some description is probably required for users to understand its purpose.

Reply:
Could you give me a use case here? I am not that familiar with the conversion you mention. Also, should we bundle this in this release or can this wait for the next patch? If it's the latter, I can create an issue and put it in Next in Line.

Reply:
Created new issue. See [here](https://github.com/jhollway/iheiddown/issues/98)